### PR TITLE
[Sprint: 53] XD-3232 Update Integration, Framework, AMQP, Batch

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/XD-3232

* Update versions
* SI TCP serializers need to be flushed since 4.2
* Remove uses of deprecated APIs
* Some modules were relying on a transitive dependency on slf4j which is no longer there
* Work around for INT-3766 (reactor-code but not reactor-stream on CP)

### CODE REVIEW

File: dependencies.properties [hunk 1]
```diff
@@ -288,34 +288,33 @@ org.slf4j\:slf4j-api=1.7.11
...
-org.springframework\:spring-orm=4.1.6.RELEASE
-org.springframework\:spring-oxm=4.1.6.RELEASE
-org.springframework\:spring-test=4.1.6.RELEASE
-org.springframework\:spring-tx=4.1.6.RELEASE
-org.springframework\:spring-web=4.1.6.RELEASE
-org.springframework\:spring-webmvc=4.1.6.RELEASE
-org.springframework\:spring-webmvc-portlet=4.1.6.RELEASE
-org.springframework\:spring-websocket=4.1.6.RELEASE
-org.springframework.amqp\:spring-amqp=1.4.5.RELEASE
-org.springframework.amqp\:spring-erlang=1.4.5.RELEASE
```
Comment (ericbottard) [inline dependencies.properties]: Is it normal that spring-erlang is gone in your version?

Comment (garyrussell) [inline dependencies.properties]: Yes, spring-erlang is no more (as of 1.5.0). It is replaced by the use of the rabbit REST API via the `RabbitManagementTemplate`.

Comment (ericbottard) [inline dependencies.properties]: Excellent, was just checking, as this jumped at me while merging.

**Comment by garyrussell** [comment]
Rebased; pushed update to s-i-kafka to fix SpEL evaluation context initialization problem.

**Comment by markpollack** [comment]
Squashed and merged as 94c44a3edd68a691ad658b0ddffb3742e1f7ea9f

**Pull request closed without merging**